### ISSUE TITLE
Implement basic slash commands

### ISSUE THREAD

**Issue by zoltan-dulac** [dylex/slack-libpurple#16]
I miss using `/me` and `/topic` from the IRC days.  They work (somewhat) on the Slack's XMPP bridge, but not with this plugin (I keep getting "The command doesn't work with this protocol").  Is this a hard thing to implement?

**Comment by dylex** [comment]
I think `/me` should be fairly straight-forward. Not sure about `/topic`. I'll look into it.

**Comment by dylex** [comment]
I found an undocumented slack api call that lets us handle all slash commands server-side.  Currently this only handles [builtin commands](https://get.slack.help/hc/en-us/articles/201259356-using-slash-commands) but theoretically could add support for any using libpurple's COMMAND_NATIVE setting.  See if this works as expected.

**Comment by zoltan-dulac** [comment]
Looks great for a lot of the command on [Slack's Slash Commands Page](https://get.slack.help/hc/en-us/articles/201259356-using-slash-commands)!  Thanks for the fix!  :-)

I did notice some strange things:

- When using `/who`, it displayed `Users here: , , , , , , , , , , , , , , , , , , , , , , , , , , ,  ... and you!` instead of showing a list of `@`-prepended user names.  I opened up the debug console and noticed that they payload for this command was something like this:

```
 {"ok":true,"response":"Users here: <@U6GDXXX0>, <@U6GDXXX1>, <@U6GDXXX2> ... and you!"}
```

(Note: I did obfuscate the `id`s for privacy reasons).

I remember when you answered [the ticket about @username support](https://github.com/dylex/slack-libpurple/issues/13) that doing lookups may be problematic.

Also, doing an `/apps bot` resulted in pidgin displaying `Here are the top three apps I found for bot` ... but without any results.  When I looked at the debug console, I got this:

```
{"username":"slackbot","text":"Here are the top three apps I found for bot","attachments":[{"text":"Connect a bot to the Slack Real Time Messaging API.","title":"Bots","id":1,"title_link":"https:\/\/slack.com\/apps\/A0F7YS25R","thumb_height":96,"thumb_width":96,"thumb_url":"https:\/\/a.slack-edge.com\/41b0a\/img\/plugins\/bot\/service_96.png","color":"4F68B0","fallback":"Bots"},{"text":"GitHub's scriptable chat bot.","title":"Hubot","id":2,"title_link":"https:\/\/slack.com\/apps\/A0F7XDU93","thumb_height":96,"thumb_width":96,"thumb_url":"https:\/\/a.slack-edge.com\/41b0a\/img\/plugins\/hubot\/service_96.png","color":"4F68B0","fallback":"Hubot"},{"text":"Spice up your team conversation with memes!","title":"Meme Bot","id":3,"title_link":"https:\/\/slack.com\/apps\/A0H25FSUS","thumb_height":96,"thumb_width":96,"thumb_url":"https:\/\/slack-files2.s3-us-west-2.amazonaws.com\/avatars\/2015-12-30\/17575614021_cdc97f6c50ea1a02f2b8_96.jpg","color":"4F68B0","fallback":"Meme Bot"},{"text":"<https:\/\/slack.com\/apps\/search?q=bot|See all results for \"bot\" in the App Directory>","id":4,"fallback":"<https:\/\/slack.com\/apps\/search?q=bot|See all results for \"bot\" in the App Directory>"}],"type":"message","subtype":"bot_message","user":"USLACKBOT","ts":"1512405311.000675","channel":"C86MEF0CB","is_ephemeral":true,"event_ts":"1512405311.000770"}
```

I am assuming it the plugin doesn't parse the JSON correctly.

In my opinion, though, I think the basic part of this ticket has been fulfilled -- basic slash commands do execute and return data.  Since these issues are specific to certain slash commands, would it be more useful for me to submit separate tickets for problems with each slash command?  Or just keep this open?

**Comment by dylex** [comment]
The `/who` fix was actually quite easy and should be better now.  The `/apps` thing should be fixed by #7, eventually. If you find more issues, feel free to file new tickets.

**Comment by zoltan-dulac** [comment]
Sounds great ... Thanks!
